Sources/SwifterSwift/SwiftStdlib/ArrayExtensions.swift

```diff
@@ -161,5 +161,34 @@ public extension Array where Element: Equatable {
             }
         }
     }
+}
 
+extension Array where Element: Hashable {
+    
+    public func filtered<E: Equatable>(duplication path: KeyPath<Element, E>) -> [Element] {
+        return reduce(into: [Element]()) { (result, e) in
+            let contains = result.contains { $0[keyPath: path] == e[keyPath: path] }
+            result += contains ? [] : [e]
+        }
+    }
+    
+    public func filtered<E: Equatable>(duplication closure: (Element) throws -> E) rethrows -> [Element] {
+        return try reduce(into: [Element]()) { (result, e) in
```

Comment:
Again, use meaningful variable names.
```suggestion
        return try reduce(into: [Element]()) { (result, element) in
```